### ISSUE TITLE
[BUG] Global teardown of Playwright takes timeout time

### ISSUE THREAD

**Issue by AlexeyCL** [microsoft/playwright#24288]
### System info
- Playwright Version: [v1.36]
- Operating System: [All, Windows 11, Ubuntu 20, macOS 13.2, etc.]
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.
Looks like global teardown takes about 30 sec, but test execution takes 1 sec. I took a simple test from CLI tools and added just the project to the config file.
Important to notice: if I execute by `testMatch =/.*.spec.ts/`, all work as required

https://github.com/AlexeyCL/Playwright-teardown

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
// playwright.config.ts
import { PlaywrightTestConfig } from '@playwright/test';
/**
 * See https://playwright.dev/docs/test-configuration.
 */
const config: PlaywrightTestConfig = {
  projects: [
    {
      name: 'test',
      testMatch: '*tests/*.spec.ts',
    },
],
  testMatch: /.*.spec.ts/,


  /* Maximum time one test can run for. */
  timeout: 60 * 1000,
  expect: {
    /**
     * Maximum time expect() should wait for the condition to be met.
     * For example in `await expect(locator).toHaveText();`
     */
    timeout: 20000,
  },
  /* Run tests in files in parallel */
  fullyParallel: !!process.env.CI,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 0 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 5 : 1,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [['list']],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    channel: process.env.DOCKER ? 'chromium' : 'chrome',
    headless: !!process.env.CI,
    viewport: { width: 1280, height: 720 },
    ignoreHTTPSErrors: true,
    baseURL: 'https://playwright.dev/',
    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
    actionTimeout: 30000,
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'retain-on-failure',
    video: 'retain-on-failure',

    screenshot: 'only-on-failure',
  },

  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
  outputDir: 'test-results/',
};

export default config;

```

**Test file (self-contained)**

```js
import { test, expect } from '@playwright/test';

test('has title', async ({ page }) => {
  await page.goto('https://playwright.dev/');

  // Expect a title "to contain" a substring.
  await expect(page).toHaveTitle(/Playwright/);
});

test('get started link', async ({ page }) => {
  await page.goto('https://playwright.dev/');

  // Click the get started link.
  await page.getByRole('link', { name: 'Get started' }).click();

  // Expects the URL to contain intro.
  await expect(page).toHaveURL(/.*intro/);
});

```

**Steps**
- [Run the test] pnpm run test
- [...]

**Expected**

[Describe expected behavior]

**Actual**

[Describe actual behavior]

**Comment by aslushnikov** [comment]
@AlexeyCL first of all, your demo project at https://github.com/AlexeyCL/Playwright-teardown does not have any global teardown.

Secondly, I ran the tests, and they passed just fine on my machine:

<img width="720" alt="image" src="https://github.com/microsoft/playwright/assets/746130/27e4dc24-5769-441e-b90f-39b38a82eb99">

So what exactly is wrong?

**Comment by AlexeyCL** [comment]
> @AlexeyCL first of all, your demo project at https://github.com/AlexeyCL/Playwright-teardown does not have any global teardown.
> 
> Secondly, I ran the tests, and they passed just fine on my machine:
> 
> <img alt="image" width="720" src="https://user-images.githubusercontent.com/746130/254537560-27e4dc24-5769-441e-b90f-39b38a82eb99.png">
> So what exactly is wrong?

Hi, I mean teardown of Playwright, I have another time execution 
<img width="668" alt="image" src="https://github.com/microsoft/playwright/assets/50070292/3e63dc95-3789-45ca-88ef-2308f9402471">

**Comment by AlexeyCL** [comment]
you can see here the time execution of tests is about 2 sec and 31 sec something else. I though about a kind of teardown of Playwright

**Comment by aslushnikov** [comment]
@AlexeyCL Ah I see, but I cannot reproduce this.

Can you please run this with `DEBUG=* npx playwright test` and paste all the logs (it'll be lots of logs!) somewhere? (e.g. to a new gist.github.com)

**Comment by AlexeyCL** [comment]
@aslushnikov hi again, I've created a [new gist ](<script src="https://gist.github.com/AlexeyCL/621cbf1ad93789a1f8699ef3fa48d83a.js"></script>)
I noticed the system stuck for a 10 seconds after these lines 
```
01b[30m+153ms\u001b[0m\n"}}} +0ms
  pw:browser [pid=42864] <gracefully close start> +153ms
  pw:test:protocol ◀ RECV {"method":"__dispatch__","params":{"method":"stdErr","params":{"text":"  \u001b[38;5;34;1mpw:protocol \u001b[0m◀ RECV {\"id\":-9999,\"result\":{}} \u001b[38;5;34m+148ms\u001b[0m\n"}}} +0ms
  pw:protocol ◀ RECV {"id":-9999,"result":{}} +148ms
```

**Comment by aslushnikov** [comment]
@AlexeyCL it looks like it takes forever to close the browser. Are you on windows by any chance? 

Can you try running your tests in different browsers and see if the issue stays:

```
npx playwright test --browser firefox
npx playwright test --browser webkit
```

**Comment by AlexeyCL** [comment]
@aslushnikov well, I played with config file and found, that issue happened in case using `channels` like here 
```
projects: [
    {
      name: 'test',
      testMatch: '*tests/*.spec.ts',
      use: {
        ...devices['Desktop Chrome'],
        channel: 'chrome'
      },
    },
```
<img width="1023" alt="image" src="https://github.com/microsoft/playwright/assets/50070292/4af70c5b-6220-472e-ad74-9007b8ee28a0">

with this config 
```  projects: [
    {
      name: 'test',
      testMatch: '*tests/*.spec.ts',
      use: {
        ...devices['Desktop Chrome'],
      },
    },
],
```
<img width="1023" alt="image" src="https://github.com/microsoft/playwright/assets/50070292/9344cd57-d2a1-481b-9cc9-afd08b1a7f9c">

**Comment by aslushnikov** [comment]
@AlexeyCL can you try launching your stable chrome and try closing it? I expect it to hang as well.

**Comment by AlexeyCL** [comment]
@aslushnikov No, I use mac. The workaround, that I found works well for me. If you tell, me that it's not an issue, we'll use a workaround

**Comment by aslushnikov** [comment]
> The workaround, that I found works well for me.

Sweet. Since I cannot reproduce this locally, lets take the workaround that works for you.

**Comment by AlexeyCL** [comment]
@aslushnikov Thank you for your quick response. Have a nice week

**Comment by aslushnikov** [comment]
@latkinsa can you please try with @playwright/test@next and see if it makes things better?

If not, then please file a separate issue with a repro that we can run locally & debug!

**Comment by latkinsa** [comment]
Thanks - it looks like there's some interference between VSCode plugins for Playwright. Clean reinstall, re-downloading browsers, and restart of VSCode seem to help.
